tests/partitions_test.py

```diff
@@ -275,6 +276,92 @@ def test_new_joiner_helps_liveness(network, args):
         network.wait_for_all_nodes_to_commit(primary=primary)
 
 
+@reqs.description("Test election while reconfiguration is in flight")
+@reqs.at_least_n_nodes(3)
+def test_election_reconfiguration(network, args):
+    # Test for issue described in https://github.com/microsoft/CCF/issues/3948
+    # Note: this test makes use of node-endorsed secondary RPC interface since
+    # new nodes never observe commit of their configuration and thus never
+    # open their service-endorsed primary RPC interface.
+    primary, backups = network.find_nodes()
+
+    LOG.info("Join new nodes without trusting them just yet")
+    new_nodes = []
+    # Start N+1 new nodes to make sure they cannot elect of them as a primary
```

Comment:
```suggestion
    # Start N+1 new nodes to make sure they cannot elect any of them as a primary
```

This comment is missing at least a word, but I think could be reworded further for clarity (I don't understand it).